apps/src/templates/progress/ProgressLesson.jsx

```diff
@@ -284,10 +289,13 @@ export default connect(state => ({
   viewAs: state.viewAs,
   showLockIcon:
     !!state.teacherSections.selectedSectionId ||
+    (state.viewAs === ViewType.Teacher &&
+      !state.stageLock.lockableAuthorized) ||
     state.viewAs === ViewType.Student,
-  lessonLockedForSection: lessonId =>
-    lessonIsLockedForAllStudents(lessonId, state),
+  lockableAuthorized: state.stageLock.lockableAuthorized,
   lessonIsVisible: (lesson, viewAs) => lessonIsVisible(lesson, state, viewAs),
+  lessonIsLockedForUser: (lesson, levels, viewAs) =>
+    lessonIsLockedForUser(lesson, levels, state, viewAs),
```

Comment:
this pattern seems unnecessarily complicated to me. we are passing in a method, then defining another method with the same name and a slightly different signature. then we are calling it just once in the render method using only props and redux state, all of which could be available to the parent. an alternate pattern which seems simpler to me would be for the parent to call this method once and pass in a boolean prop indicating whether this component is locked.

I get that you are just copying the existing pattern here, and fixing this could be a bit of work because the parent is not currently connected to redux. so, just flagging this as a pattern to avoid in the future when possible. interested if others agree or disagree.